cmd/kube-proxy/app/server.go

```diff
@@ -76,6 +80,38 @@ func NewProxyServer() *ProxyServer {
 		ResourceContainer:  "/kube-proxy",
 		SyncPeriod:         5 * time.Second,
 	}
+
+	return &s
+}
+
+// NewProxyServerCommand creates a new ProxyServer object with default parameters
+func NewProxyServerCommand() *cobra.Command {
+	s := NewProxyServer()
+
+	cmd := &cobra.Command{
+		Use:   "kube-proxy",
+		Short: "The kube-proxy implements the code needed to support kubernetes services",
+		Long: `The Kubernetes network proxy runs on each node. This
+reflects services as defined in the Kubernetes API on each node and can do simple
+TCP,UDP stream forwarding or round robin TCP,UDP forwarding across a set of backends.
+Service cluster ips and ports are currently found through Docker-links-compatible
+environment variables specifying ports opened by the service proxy. There is an optional
+addon that provides cluster DNS for these cluster IPs. The user must create a service
+with the apiserver API to configure the proxy.`,
+		Run: func(_ *cobra.Command, args []string) {
+			if err := s.Run(args); err != nil {
+				glog.Fatalf("%v", err)
+			}
+		},
+	}
+	genutils.AddGeneratedDocsCommands(cmd)
+
+	s.AddFlags(cmd.Flags())
+	pflag.CommandLine.AddGoFlagSet(goflag.CommandLine)
+	cmd.SetGlobalNormalizationFunc(util.WordSepNormalizeFunc)
```

Comment:
Still used by contrib/mesos and integration. maybe I can keep working on those. Mesos really needs to make the switch to get sanity on their flags...  But separate PR